### PR TITLE
feat-film-list-selector

### PR DESCRIPTION
**Pull request by yurcco**

### CODE REVIEW

File: src/app/components/film-selected/film-selected.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<p>film-selected works!</p>
+<div *ngIf="film.name">
+    <h1>{{film.name}}</h1>
+    <p>{{film.description}}</p>
+    <img src="{{film.img}}" alt="film image">
+  </div>
```
Comment (MTrofimenko) [inline src/app/components/film-selected/film-selected.component.html]: format please

File: src/app/components/film-selected/film-selected.component.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { Film } from '../../models/film';
+@Component({
+  selector: 'app-film-selected',
+  templateUrl: './film-selected.component.html',
+  styleUrls: ['./film-selected.component.css']
+})
+export class FilmSelectedComponent implements OnInit {
+  @Input() film: Film;
+  
+  constructor() { }
```
Comment (MTrofimenko) [inline src/app/components/film-selected/film-selected.component.ts]: Can be deleted

File: src/app/store/reducers/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import * as fromFilms from "./films";
+export interface State {
+  films: fromFilms.State;
+}
+export const reducers: ActionReducerMap<State> = {
+  films: fromFilms.reducer
+};
+export function logger(reducer: ActionReducer<State>): ActionReducer<State> {
+  return function(state: State, action: any): State {
+    console.log("state", state);
```
Comment (MTrofimenko) [inline src/app/store/reducers/index.ts]: Do you need these logs?

File: src/app/app.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  selector: "app-root",
+  templateUrl: "./app.component.html",
+  styleUrls: ["./app.component.css"]
+})
+export class AppComponent {
+  films$: Observable<Film[]>;
+  selected$: Observable<any>;
+
+  constructor(private store: Store<fromRoot.State>) {
+    this.films$ = store.select(fromRoot.getAllFilms);
```
[the remark this answers is not in the document]
Comment (yurcco) [reply src/app/app.component.ts]: done

File: src/app/components/film-item/film-item.component.html [hunk 5]
```diff
@@ -0,0 +1 @@
+<li (click)="select.emit(film.id)">{{ film.name }}</li>
```
[the remark this answers is not in the document]
Comment (yurcco) [reply src/app/components/film-item/film-item.component.html]: done

File: src/app/components/film-list/film-list.component.html [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<div class="list">
+  <h2>{{ label }}</h2>
+  <app-film-item
+    *ngFor="let film of films"
+    [film]="film"
+    (select)="select.emit($event)"
```
[the remark this answers is not in the document]
Comment (yurcco) [reply src/app/components/film-list/film-list.component.html]: done

File: src/app/components/film-list/film-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  selector: "app-film-list",
+  templateUrl: "./film-list.component.html",
+  styleUrls: ["./film-list.component.css"]
+})
+export class FilmListComponent implements OnInit, OnChanges {
+  @Input() films: Film[];
+  @Input() label: string;
+  @Output() select = new EventEmitter();
+  constructor() {}
+  ngOnChanges(changes: SimpleChanges): void {}
```
[the remark this answers is not in the document]
Comment (yurcco) [reply src/app/components/film-list/film-list.component.ts]: done

File: src/app/components/film-selected/film-selected.component.html (hunk 1 above)
Comment (yurcco) [reply src/app/components/film-selected/film-selected.component.html]: done

File: src/app/components/film-selected/film-selected.component.ts (hunk 2 above)
Comment (yurcco) [reply src/app/components/film-selected/film-selected.component.ts]: done

File: src/app/store/reducers/index.ts (hunk 3 above)
Comment (yurcco) [reply src/app/store/reducers/index.ts]: I don't, but in the example  it was for see haw changing our store

**Pull request merged**